### ISSUE TITLE
Don't flag both versions of duplicate albums as duplicate

### ISSUE THREAD

**Issue by raffomania** [sampsyo/beets#1265]
When reimporting (read: retagging) my library that contains a duplicate album, beet flags both versions of the album as duplicate, thus asking me twice what I want to do about it. Expected behaviour: If I remove the first version, don't flag the second version as duplicate.

**Comment by sampsyo** [comment]
Hi there—can you be slightly more specific about the steps here? Something like this:

1. Start with two copies of the same album, A and B.
2. Import A.
3. Import B. When prompted about duplicates, say "keep both".
4. Re-import everything.
5. When prompted about duplicates for A, say "remove old".
6. When prompted about duplicates for B… what happens here?

**Comment by Freso** [comment]
> When prompted about duplicates for B… what happens here?

As I understand the original comment, the issue is that it prompts about duplicates for B at all, once A is gone, so it should rather be something like:

1. Start with two copies of the same album, A and B.
2. Import A.
3. Import B. When prompted about duplicates, say "keep both".
4. Re-import everything.
5. When prompted about duplicates for A, say "remove old".
6. Doesn't prompt for B, as B is no longer a duplicate.

**Comment by sampsyo** [comment]
Thanks, @Freso. If that's the case, I'm not sure if we should do anything. In step 4, the user ran a command that asked for A and B to be imported. So even though B was removed in step 5, the user asked beets to import it again in step 6! And since A is in the library, the duplicate detector gets triggered.

Changing this behavior might actually make the behavior *more* confusing.

**Comment by raffomania** [comment]
@Freso, thanks for the clarification, that was exactly what I meant.
An additional fact regarding my special case was that I imported the albums without running the autotagger, which resulted in beets not asking me about the duplicates.

@sampsyo, IMO all this confusion stems out of the strong connection between importing and autotagging - in my use case, all I ever wanted to do in step 4-6 was to clean my library up, remove duplicates etc (by running the autotagger), but I never wanted to *import* anything.

**Comment by sampsyo** [comment]
I see that—if your workflow is to re-auto-tag everything, I definitely see how B seems like it has been "resurrected". Perhaps the solution is to use the duplicates plugin first before running the re-import? As it stands, I can't think of a clean way to resolve this confusion that doesn't break other promises for the `import` command—namely, that it will autotag everything you ask it to, in order.

**Comment by raffomania** [comment]
Well, the duplicate plugin certainly is one possible solution.
It still bugs me that this is kind of an unexpected behavior that can have grave consequences. I lost many audio files because of this :( Maybe we could arrange a warning that is displayed when a duplicate album is not in the file system anymore?

**Comment by sampsyo** [comment]
Yeah, you're totally right that its unexpected when you're doing a "re-import everything to autotag the library" workflow. I'm just not quite sure what the right fix is. A nice, clean idea for a solution here would be very welcome. I'm going to leave this open for now while we brainstorm.

**Comment by raffomania** [comment]
Well, what about a general warning when tagging non-existent albums? That would definitely help, I think.
